soroban-sdk/src/env.rs

```diff
@@ -419,6 +420,26 @@ impl Env {
         }
     }
 
+    /// Authorizes sub-contract calls on behalf of the current contract.
+    ///
+    /// All the direct calls that the current contract performs are always
+    /// considered to have been authorized. This is only needed to authorize
+    /// deeper calls that originate from the next contract call from the current
+    /// contract.
+    ///
+    /// For example, if the contract A calls contract B, contract
+    /// B calls contract C and contract C calls `A.require_auth()`, then an
+    /// entry corresponding to C call has to be passed in `auth_entries`. It
+    /// doesn't matter if contract B called `require_auth` or not. If contract A
+    /// calls contract B again, then `authorize_as_curr_contract` has to be
+    /// called again with the respective entries.
+    ///
+    ///
+    pub fn authorize_as_curr_contract(&self, auth_entries: Vec<InvokerContractAuthEntry>) {
```

Comment:
Let's avoid the term `curr`.

I think we can simply name this `authorize`, because the only time a contract would authorize anything is when the contract is the one doing the authorization?

```suggestion
    pub fn authorize(&self, auth_entries: Vec<InvokerContractAuthEntry>) {
```

Alternatively, if we would like to be more specific, I think we can say:

```suggestion
    pub fn authorize_as_self(&self, auth_entries: Vec<InvokerContractAuthEntry>) {
```